### PR TITLE
Kvm create virtio-blk unit tests

### PR DESCRIPTION
**Pull request by artek-koltun**

### CODE REVIEW

File: pkg/frontend/frontend.go [hunk 1]
```diff
@@ -14,8 +14,10 @@ import (
 )
 
 var (
```
Comment (glimchb) [inline pkg/frontend/frontend.go:16]: looks like those are not just frontend errors and can appear everywhere, consider moving them to utils.go or jsonrpc.go

File: pkg/kvm/kvm_test.go [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+			}
+			*resultDeleteVirtioBLk = models.VhostDeleteControllerResult(true)
+		}
+		return s.err
+	} else {
+		return s.err
+	}
+}
+
+type mockCall struct {
```
Comment (glimchb) [inline pkg/kvm/kvm_test.go:75]: have you considered using https://github.com/vektra/mockery instead ?

File: pkg/kvm/kvm_test.go [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+func (s *mockQmpServer) write(data string, conn net.Conn) {
+	log.Println("QMP server send:", data)
+	_, err := conn.Write([]byte(data))
+	if err != nil {
+		log.Panicf("QMP server failed to write: %v", data)
+	}
+}
+
+func (s *mockQmpServer) read(conn net.Conn) string {
+	buf := make([]byte, 512)
```
Comment (glimchb) [inline pkg/kvm/kvm_test.go:241]: looks like this is new style of testing, please consider either change the rest of tests to look the same or keep the old style... it is hard to maintain project that has a lot of coding styles in it..

**Review by glimchb — CHANGES_REQUESTED** [review]
few questions inside

File: pkg/kvm/kvm_test.go (hunk 3 above)
Comment (glimchb) [reply pkg/kvm/kvm_test.go:241]: from slack: I'm referring to `Expect*` calls
I'm not against it, I just looking for the same look and feel across the project

File: pkg/frontend/frontend.go (hunk 1 above)
Comment (artek-koltun) [reply pkg/frontend/frontend.go]: moved to `jsonrpc.go`

File: pkg/kvm/kvm_test.go (hunk 2 above)
Comment (artek-koltun) [reply pkg/kvm/kvm_test.go:75]: As I know, all go mock frameworks require code generation and I'd avoid it. In addition as I see other tests are leveraging a goroutine listening on unix socket and sending stub responses and wanted to stick to that approach exercising the full path.

Comment (glimchb) [reply pkg/kvm/kvm_test.go:75]: ok, just checking

File: pkg/kvm/kvm_test.go (hunk 3 above)
Comment (glimchb) [reply pkg/kvm/kvm_test.go:241]: sure

**Review by glimchb — APPROVED** [review]
lgtm

**Comment by artek-koltun** [comment]
> @artek-koltun tests failed on
> 
> ```
> 2023-03-16T12:45:55.4596984Z     --- FAIL: TestDeleteVirtioBlk/spdk_failed_to_delete_virtio-blk (0.00s)
> ```
> 
> due to
> 
> ```
> 2023-03-16T12:45:55.4576875Z 2023/03/16 12:45:55 DeleteVirtioBlk: Received from client: name:"virtio-blk-42"
> 2023-03-16T12:45:55.4577172Z 2023/03/16 12:45:55 error: stub error
> 2023-03-16T12:45:55.4577555Z 2023/03/16 12:45:55 Error running underlying cmd on opi-spdk bridge: stub error
> 2023-03-16T12:45:55.4578298Z     kvm_test.go:434: Not all expected calls were performed
> ```

thanks for looking into the logs.
Apparently, race condition for `callIndex` member due to `-race` test option. Need some time to investigate and adjust. Pls do not merge

**Comment by artek-koltun** [comment]
> per @artek-koltun request waiting with this until race condition is addressed
`-race` adds some random delays to execution flow. Put a mutex around sensitive operations and ran the tests 50_000 times with no failures. Suppose we can merge the patch.

**Review by glimchb — APPROVED** [review]
thanks for writing tests and fixing race condition!

**Pull request merged**